webapp/src/actions/index.js

```diff
@@ -80,4 +80,34 @@ export function performSearch(terms) {
 
         return dispatch(searchPostsWithParams(teamId, {terms, is_or_search: false, include_deleted_channels: viewArchivedChannels, page: 0, per_page: 20}, true));
     };
-}
\ No newline at end of file
+}
+
+export function requeueItem(itemId) {
+    return async (dispatch, getState) => {
+        const command = `/agenda requeue ${itemId}`;
+        await clientExecuteCommand(dispatch, getState, command);
+        return {data: true};
+    };
+}
+
+export async function clientExecuteCommand(dispatch, getState, command) {
+    let currentChannel = getCurrentChannel(getState());
+    const currentTeamId = getCurrentTeamId(getState());
+
+    // Default to town square if there is no current channel (i.e., if Mattermost has not yet loaded)
+    if (!currentChannel) {
+        currentChannel = await Client4.getChannelByName(currentTeamId, 'town-square');
+    }
```

Comment:
Mattermost has loaded if we are running this command right?

Reply:
**(R1)**Most of this function snippet is borrowed from https://github.com/larkox/mattermost-plugin-badges/blob/709bf26a2700a7b1830d9dd436e30d0fc08bd187/webapp/src/actions/actions.ts#L105

Reply:
should we skip this check?